app/src/main/java/com/example/marvelapp/data/mapper/CharacterRealmMapper.kt

```diff
@@ -15,34 +15,82 @@ import com.example.marvelapp.domain.entity.StoriesEntity
 import com.example.marvelapp.domain.entity.ThumbnailEntity
 import com.example.marvelapp.domain.entity.UrlsEntity
 import io.realm.RealmList
+import io.realm.RealmResults
 
 object CharacterRealmMapper {
 
-    private fun transformDBCharacter(characterEntity: CharacterEntity): CharacterRealmEntity =
-        CharacterRealmEntity(
-            characterEntity.id,
-            characterEntity.name,
-            characterEntity.description,
-            characterEntity.modified,
-            transformDBThumbnail(characterEntity.thumbnail),
-            characterEntity.resourceUri,
-            transformDBComics(characterEntity.comics),
-            transformDBEvents(characterEntity.events),
-            transformDBStories(characterEntity.stories),
-            transformDBSeries(characterEntity.series),
-            transformDBListOfUrls(characterEntity.urls)
+    private fun transformDBCharacter(characterEntity: CharacterEntity): CharacterRealmEntity = CharacterRealmEntity(
+        characterEntity.id,
+        characterEntity.name,
+        characterEntity.description,
+        characterEntity.modified,
+        transformDBThumbnail(characterEntity.thumbnail),
+        characterEntity.resourceUri,
+        transformDBComics(characterEntity.comics),
+        transformDBEvents(characterEntity.events),
+        transformDBStories(characterEntity.stories),
+        transformDBSeries(characterEntity.series),
+        transformDBListOfUrls(characterEntity.urls)
+    )
+
+    private fun transformCharacterEntityToRealm(characterRealmEntity: CharacterRealmEntity): CharacterEntity =
+        CharacterEntity(
+            characterRealmEntity.id,
+            characterRealmEntity.name,
+            characterRealmEntity.description,
+            characterRealmEntity.modified,
+            transformDBThumbnailr(characterRealmEntity.thumbnail!!),
```

Comment:
DO NOT USE null assert !!

Reply:
Do you know the difference between **!!** and **?.** ?

Reply:
The operator?, will verify if a variable is null before doing any operation with it, while the operator !!,  apply the operation without doing any verification

Reply:
great!